code/game/objects/items/stacks/stack_recipe.dm

```diff
@@ -40,6 +40,13 @@
 		R.update_icon()
 	..()
 
+/datum/stack_recipe/bluespacecrystals
+/datum/stack_recipe/bluespacecrystals/post_build(var/obj/item/stack/S, var/obj/result)
+	if(istype(result, obj/item/weapon/ore/bluespacecrystal/B))
```

Comment:
You need a `/` preceding `obj/item/...`, and no `B` at the end, since this is a type path, and not a variable.